### PR TITLE
[Julia] pass optimize to typed

### PR DESCRIPTION
**Pull request by vchuravy**
`code_typed` in Julia also takes the optimize option.

### CODE REVIEW

**Comment by mattgodbolt** [comment]
Can you explain why this change is handy please? I don't know anything about Julia :)

**Review by mattgodbolt — APPROVED** [review]

**Comment by vchuravy** [comment]
:) multi-stage compiler and the LLVM and typed levels have pre & post-optimization modes.

File: etc/scripts/julia_wrapper.jl [hunk 1]
```diff
@@ -84,17 +84,17 @@ open(output_path, "w") do io
     for (me_fun, me_types, me) in m_methods
         io_buf = IOBuffer() # string buffer
         if format == "typed"
-            ir, retval = InteractiveUtils.code_typed(me_fun, me_types, debuginfo=debuginfo)[1]
+            ir, retval = InteractiveUtils.code_typed(me_fun, me_types; optimize, debuginfo)[1]
             Base.IRShow.show_ir(io_buf, ir)
         elseif format == "lowered"
-            cl = Base.code_lowered(me_fun, me_types, debuginfo=debuginfo)
+            cl = Base.code_lowered(me_fun, me_types,; debuginfo)
```
Comment (jeremy-rifkin) [inline etc/scripts/julia_wrapper.jl:90]: Pardon me as I don't know Julia (I've been meaning to learn it), is `,;` correct syntax here?

**Review by jeremy-rifkin — APPROVED** [review]
Looks good to me other than the one little syntax question

File: etc/scripts/julia_wrapper.jl (hunk 1 above)
Comment (vchuravy) [reply etc/scripts/julia_wrapper.jl:90]: It's fine, but cleaner would have been the single `;`. I.e. `,` is a trailing comma.

```
julia> f(a,; kwargs...) = a, kwargs
f (generic function with 1 method)

julia> f(1; b=2)
(1, Base.Pairs(:b => 2))
```

and `;` denotes the beginning of keyword-arguments.

Comment (vchuravy) [inline etc/scripts/julia_wrapper.jl:90]:
```suggestion
            cl = Base.code_lowered(me_fun, me_types; debuginfo)
```

**Pull request merged**